designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputFileResource.kt

```diff
@@ -0,0 +1,121 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Cancel
+import androidx.compose.material.icons.outlined.FileUpload
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import org.hisp.dhis.mobile.ui.designsystem.component.UploadFileState.ADD
+import org.hisp.dhis.mobile.ui.designsystem.component.UploadFileState.LOADED
+import org.hisp.dhis.mobile.ui.designsystem.component.UploadFileState.UPLOADING
+import org.hisp.dhis.mobile.ui.designsystem.resource.provideStringResource
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+
+@Composable
+fun InputFileResource(
+    title: String,
+    fileName: MutableState<String?> = mutableStateOf(null),
+    fileWeight: MutableState<String?> = mutableStateOf(null),
+    onSelectFile: () -> Unit,
+    onUploadFile: () -> Unit,
+    onClear: () -> Unit = {},
+    uploadFileState: UploadFileState = ADD,
+) {
+    var currentState by remember {
+        mutableStateOf(uploadFileState)
+    }
+
+    val primaryButton: @Composable (() -> Unit)? = if (currentState == LOADED) {
+        {
+            IconButton(
+                modifier = Modifier.testTag("INPUT_FILE_RESOURCE_CLEAR_BUTTON"),
+                icon = {
+                    Icon(
+                        imageVector = Icons.Outlined.Cancel,
+                        contentDescription = "Icon Button",
+                    )
+                },
+                onClick = {
+                    currentState = ADD
+                    onClear.invoke()
+                },
+            )
+        }
+    } else {
+        null
+    }
+
+    val secondaryButton: @Composable (() -> Unit)? =
+        if (currentState == LOADED) {
+            {
+                SquareIconButton(
+                    icon = {
+                        Icon(
+                            imageVector = Icons.Outlined.FileUpload,
+                            contentDescription = "Upload Icon Button",
+                        )
+                    },
+                ) {
+                    currentState = UPLOADING
+                    onUploadFile.invoke()
+                }
+            }
+        } else {
+            null
+        }
+
+    InputShell(
+        title,
+        state = if (currentState == UPLOADING) InputShellState.FOCUSED else InputShellState.UNFOCUSED,
+        inputField = {
+            when (currentState) {
+                ADD -> {
+                    Button(
+                        modifier = Modifier.padding(end = Spacing.Spacing16).fillMaxWidth(),
+                        style = ButtonStyle.ELEVATED,
+                        text = provideStringResource("add_file"),
```

Comment:
remember to add the add file string as a parameter so that we can manage translations from the app, and the default value can be that